clients/admin-ui/src/features/common/system-data-flow/DataFlowAccordionForm.tsx

```diff
@@ -0,0 +1,193 @@
+import {
+  AccordionButton,
+  AccordionIcon,
+  AccordionItem,
+  AccordionPanel,
+  Button,
+  ButtonGroup,
+  Flex,
+  Spacer,
+  Stack,
+  Tag,
+  Text,
+  useDisclosure,
+  useToast,
+} from "@fidesui/react";
+import { FormGuard } from "common/hooks/useIsAnyFormDirty";
+import { GearLightIcon } from "common/Icon";
+import QuestionTooltip from "common/QuestionTooltip";
+import { DataFlowSystemsDeleteTable } from "common/system-data-flow/DataFlowSystemsDeleteTable";
+import DataFlowSystemsModal from "common/system-data-flow/DataFlowSystemsModal";
+import { successToastParams } from "common/toast";
+import { Form, Formik } from "formik";
+import React, { useEffect, useMemo, useState } from "react";
+
+import {
+  useGetAllSystemsQuery,
+  useUpdateSystemMutation,
+} from "~/features/system";
+import { DataFlow, System } from "~/types/api";
+
+const defaultInitialValues = {
+  dataFlowSystems: [] as DataFlow[],
+};
+
+export type FormValues = typeof defaultInitialValues;
+
+type DataFlowAccordionItemProps = {
+  isIngress?: boolean;
+  system: System;
+  isSystemTab?: boolean;
+};
+
+export const DataFlowAccordionForm = ({
+  system,
+  isIngress,
+  isSystemTab,
+}: DataFlowAccordionItemProps) => {
+  const toast = useToast();
+  const flowType = isIngress ? "Source" : "Destination";
+  const pluralFlowType = `${flowType}s`;
+  const dataFlowSystemsModal = useDisclosure();
+  const [updateSystemMutationTrigger] = useUpdateSystemMutation();
+
+  const { data } = useGetAllSystemsQuery();
+  const systems = useMemo(() => data || [], [data]);
```

Comment:
oh this is actually a great case to use the selector `selectAllSystems`

```
// set up the subscription
useGetAllSystemsQuery();

// access the data
const systems = useAppSelector(selectAllSystems);
```

I'd actually say `selectAllSystems` should probably return an empty array if there's no data. that would simplify some code elsewhere, too.

Reply:
great idea! I went ahead and updated the selection to return an empty array along with using the selector instead

Reply:
That seems like a good pattern to follow going forward. It makes component logic easier to follow when `undefined` never shows up